omc3/optics_measurements/rdt.py

```diff
@@ -59,14 +83,22 @@ def calculate(measure_input, input_files, tunes, phases, invariants, header):
         for_rdts = _best_90_degree_phases(meas_input, bpm_names, phases, tunes, plane)
         LOGGER.info(f"Average phase advance between BPM pairs: {for_rdts.loc[:,'MEAS'].mean()}")
         for rdt in SINGLE_PLANE_RDTS[plane]:
-            df = _process_rdt(meas_input, input_files, for_rdts, invariants, plane, rdt)
+            try:
+                df = _process_rdt(meas_input, input_files, for_rdts, invariants, plane, rdt)
+            except ValueError as e:  # catch the AMP line not being found in the lin file
+                LOGGER.warning(f"RDT calculation failed for {jklm2str(*rdt)}: {str(e)}")
+                continue
             write(df, add_freq_to_header(header, plane, rdt), meas_input, plane, rdt)
```

Comment:
Shouldn't the write line be in the try block? As I see it, a failure in `_process_rdt` will log a warning but still try to write to disk, which would fail.

Reply:
good catch, I just moved them